apiary.apib

```diff
@@ -71,6 +71,111 @@ The following is an example error response:
+    ```
+
+2. Append this token to your Unmade Editor url, for example:
+
+    ```
+    https://{your-partner-subdomain}.embed.unmade.com/v1/products/{your-product}/?init=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyIjoxNTE2MjM5MDIyfQ.66pC0Ncp6ZiUGljv5dlF8uKO1jQQ04Fvkj7z56VfBjg
+    ```
+
+3. The Unmade will decode this token and
+
+    If the token IS valid the Unmade Editor UI will enable the ability to upload a Logo and process, save that file.
+
+    If the token is NOT valid the error will be logged and the Unmade Editor UI will disable the functionality of uploading content.
+
+    We have the following types of validation/error messages:
+    - wrong key
+    - unable to decipher
+    - missing info in the payload
+    - missing "user" property on the payload
+
+
+### Initialising an Unmade Editor without a unique identifier for an authenticated user
+
+If you initialise the Unmade Editor without an authenticated user then the user will still be able to customise the product and can be prompted to authenticate with your e-commerce site if and when they try to upload a logo.
+
+1. For an unauthenticated user, the Unmade Editor should be initialised without a JWT payload.
+
+    ```
+    https://{your-partner-subdomain}.embed.unmade.com/v1/products/{your-product}/
+    ```
+
+2. If and when the unauthenticated user clicks to upload a logo, the Unmade Editor iFrame will:
+
+    Save the design and then inform your e-commerce site that the user needs to be authenticated in order to continue and send a JavaScript postMessage() to the parent window requesting the log in action and pass along the Design ID. The payload of the postmessage will look something like this:
+
+    ```
+    {
+      type: "modal",
+      payload: {
+        show: "login",
+        params: {
+          designID: "7fad56a8def33980cc3f90"
+        }
+      }
+    }
+    ```
+
+3. Your e-commerce site will need to listen for this JavaScript postMessage() and when it is received direct the user to authenticate.
+
+    Once your e-commerce site has authenticated the user it can re-initialise the Unmade Editor with a unique identifier for the authenticated user and the saved design id and the user will be able to continue with uploading their logo. For example
+
+    ```
+    https://{your-partner-subdomaini}.embed.unmade.com/v1/products/{your-product}/?init=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyIjoxNTE2MjM5MDIyfQ.66pC0Ncp6ZiUGljv5dlF8uKO1jQQ04Fvkj7z56VfBjg&design_id=7fad56a8def33980cc3f90
+    ```
+
+4. The Unmade Editor will then enable the authenticated user to upload logos and continue editing their design.
+
+
+### Restrictions for user uploaded logos
+
+The Unmade platform will perform some checks on the files that are uploaded. The file upload restrictions are:
+
+- File size limit 2MB
+- SVG file type with no script elements, no embedded images or text elements
+- Single colour, no gradients or patterns fills
```

Comment:
Minor details, but my understanding is:
- The max file size is 1MB and a smaller file size can be configured if desired.
- We allow up to 6 colours in the file.

Reply:
@marcharlier I've made these changes - if there is anyone I need to check this with then please say

Reply:
That limit is defined [https://github.com/unmadeworks/embed-service/blob/01ce696f8a26ff56d3289263b9c67dd3b08d76ed/embed_service/embed/settings.py#L529).

That setting can be overriden per embed-service.

Reply:
I'll expand these requirements to include more exact info about what we mean by "colours", and exactly which tags are disallowed.

Mainly because I don't think we've got it actually written up anywhere, and it would be useful for us primarily!